opendebates/models.py

```diff
@@ -45,20 +46,51 @@ class SiteMode(CachingMixin, models.Model):
     announcement_link = models.URLField(null=True, blank=True)
     announcement_page_regex = models.CharField(max_length=255, null=True, blank=True)
 
-    banner_header_title = models.TextField(default=u'Welcome to the<br>Open Debate')
-    banner_header_copy = models.TextField(default=u'Ask about the issues that are most '
-                                          'important to you -- then vote for other '
-                                          'important questions and encourage friends '
-                                          'to do the same!')
+    hashtag = models.CharField(default=site_defaults.HASHTAG, max_length=255)
 
-    popup_after_submission_text = models.TextField(default=u'Next, help your question '
-                                                   'collect votes. Share it on social '
-                                                   'media and email it to friends.')
+    banner_header_title = models.TextField(default=site_defaults.BANNER_HEADER_TITLE)
+    banner_header_copy = models.TextField(default=site_defaults.BANNER_HEADER_COPY)
+
+    popup_after_submission_text = models.TextField(
+        default=site_defaults.POPUP_AFTER_SUBMISSION_TEXT
+    )
+
+    email_subject = models.CharField(default=site_defaults.EMAIL_SUBJECT, max_length=998)
+    email_body = models.TextField(default=site_defaults.EMAIL_BODY)
+
+    facebook_image = models.URLField(default=site_defaults.FACEBOOK_IMAGE)
+    facebook_question_description = models.TextField(
+        default=site_defaults.FACEBOOK_QUESTION_DESCRIPTION
+    )
+    facebook_question_title = models.TextField(default=site_defaults.FACEBOOK_QUESTION_TITLE)
+    facebook_site_description = models.TextField(default=site_defaults.FACEBOOK_SITE_DESCRIPTION)
+    facebook_site_title = models.TextField(default=site_defaults.FACEBOOK_SITE_TITLE)
+
+    twitter_image = models.URLField(default=site_defaults.TWITTER_IMAGE)
+    twitter_question_description = models.TextField(
+        default=site_defaults.TWITTER_QUESTION_DESCRIPTION
+    )
+    twitter_question_text = models.TextField(default=site_defaults.TWITTER_QUESTION_TEXT)
+    twitter_question_title = models.TextField(default=site_defaults.TWITTER_QUESTION_TITLE)
+    twitter_site_description = models.TextField(default=site_defaults.TWITTER_SITE_DESCRIPTION)
+    twitter_site_text = models.TextField(default=site_defaults.TWITTER_SITE_TEXT)
+    twitter_site_title = models.TextField(default=site_defaults.TWITTER_SITE_TITLE)
 
     objects = CachingManager()
 
 
-class Submission(models.Model):
+class CachedSiteModeMixin(object):
+
+    @property
+    def site_mode(self):
+        # avoid lots of unneeded round trips to memcached in production
```

Comment:
Good catch! I would like to avoid caching anything on a Python object that may persist in the interpreter outside the request/response cycle (as doing so may lead to some difficult to troubleshoot invalidation issues), however, we already have the site mode in the template context so it's just a matter of figuring out how to get it into these `Submission` model methods (or how to refactor them/put them elsewhere so that's not required).

For the time being, I added a template tag which pre-fills `_cached_site_mode` with the `SITE_MODE` available in the template context, along with a test to make sure the link doesn't get broken.

Longer term, this probably needs some refactoring, but I think this will work for now.